frontend/model/contracts/identity.js

```diff
@@ -202,13 +202,13 @@ sbp('chelonia/defineContract', {
       }),
       async process ({ hash, data }, { state }) {
         const { groupContractID, inviteSecret } = data
-        if (has(state.groups, groupContractID)) {
+        if (has(state.groups, groupContractID) && state.groups[groupContractID].active) {
           throw new Error(`Cannot join already joined group ${groupContractID}`)
         }
 
         const inviteSecretId = await sbp('chelonia/crypto/keyId', new Secret(inviteSecret))
 
-        state.groups[groupContractID] = { hash, inviteSecretId }
+        state.groups[groupContractID] = { hash, inviteSecretId, active: true }
```

Comment:
This is a rather huge / significant change.

What is this `active` thing, why is it being added, and doesn't this have the potential to break all existing groups that have been created and joined and do not have this property?

Reply:
For correctness, `state.groups[groupContractID]` is no longer deleted. This is needed for the leave action to work correctly, as the `reference` property needs to be available, which it won't be if the state is removed.

> doesn't this have the potential to break all existing groups that have been created and joined and do not have this property?

Yes, but I'd say all of the changes in this PR are breaking. Thanks for pointing that out.

Reply:
I think the easiest change right now is to make this a negative property (e.g., `inactive` / `left`) so that it doesn't break existing groups.

`'state/vuex/postUpgradeVerification'` seems inappropriate to solve this because:

  * It deals with Vuex state instead of contract state
  * It is executed in the browsing context. This would need to run in the service worker.
  * It is entirely unaware of contract versions. Doing this correctly needs awareness of contract versions.

Implementing #2108 would address this to some extent. However, I think that the preferred solution to this would require a solid in-contract upgrade mechanism. Looking at how contracts are handled now, I see several deficiencies:

1. It don't think versions are checked at all, which means that someone could downgrade a contract. This should probably be prevented.
2. Because contracts can be downgraded, they need to be forward and backwards compatible.

Reply:
> It don't think versions are checked at all, which means that someone could downgrade a contract. This should probably be prevented.

Yep there's #1957 to address exactly that.

> Because contracts can be downgraded, they need to be forward and backwards compatible.

It's fine to assume contracts will never be downgraded in the sense that contracts should never be downgraded and we will never downgrade them. If someone hacks the app to do this they are responsible for any app breakage, not us.

Reply:
See new commit addressing this by using a negated property.